tests/unit/stacked-vertical-bar-test.js

```diff
@@ -0,0 +1,773 @@
+          expectedSliceHeights[iSlice]),
+        "The bar for '" + barLabel +
+          "' has a slice with height " + expectedSliceHeights[iSlice] +
+          " px out of " + barHeight +
+          " px (actual height: " + sliceHeights[iSlice] +
+          " px)");
+    }
+  }
+});
+
+test('The bars have the correct heights relative to each other', function(assert) {
+  var dataByBarLabel, grossBarSums, minGrossBarSum, expectedBarHeightRatios,
+    actualBarHeights, minActualBarHeight, barLabel, expectedBarHeight;
+
+  dataByBarLabel = _.groupBy(three_ranges, 'barLabel');
+  assert.expect(_.keys(dataByBarLabel).length);
+
+  // Compute the expected heights of each bar as a percentage
+  // of the expected height of the shortest bar.
+  grossBarSums = _.mapValues(dataByBarLabel, function(barData) {
+    var sum = 0.0;
+    for (var iDatum = 0; iDatum < barData.length; iDatum++) {
+      sum += Math.abs(barData[iDatum].value);
+    }
+    return sum;
+  });
+  minGrossBarSum = _.min(_.values(grossBarSums));
+  expectedBarHeightRatios = _.mapValues(grossBarSums, function(sum) {
+    return (sum / minGrossBarSum);
+  });
+
+  this.subject({data: three_ranges});
+  this.render();
+
+  // Find the height of each SVG bar by summing the heights of its slices,
+  // as well as the shortest height.
+  actualBarHeights = {};
+  this.$('svg g.bars').each(function() {
+    var slices = $('rect', this);
+    actualBarHeights[$(this).text()] = _.sum(_.map(slices, getHeightOfSvgSlice));
+  });
+  minActualBarHeight = _.min(_.values(actualBarHeights));
+
+  // Compare the expected and actual heights
+  // and assert that they are approximately equal.
+  for (barLabel in expectedBarHeightRatios) {
+    expectedBarHeight = minActualBarHeight * expectedBarHeightRatios[barLabel];
+    assert.ok(
+      equalsWithTolerance(
+        actualBarHeights[barLabel],
+        expectedBarHeight),
+      "The bar for '" + barLabel +
+        "' has a height of " + expectedBarHeight +
+        " px (actual height: " + actualBarHeights[barLabel] +
+        " px)");
+  }
+});
+
+test('The bars have the correct heights relative to the values on the y-axis ticks', function(assert) {
+  var dataByBarLabel, grossBarSums, pixelsPerDataUnit, expectedBarHeights,
+    actualBarHeights, barLabel;
+
+  dataByBarLabel = _.groupBy(three_ranges, 'barLabel');
+  assert.expect(1 + _.keys(dataByBarLabel).length);
+
+  // Compute the expected heights of each bar as a percentage
+  // of the expected height of the shortest bar.
+  grossBarSums = _.mapValues(dataByBarLabel, function(barData) {
+    var sum = 0.0;
+    for (var iDatum = 0; iDatum < barData.length; iDatum++) {
+      sum += Math.abs(barData[iDatum].value);
+    }
+    return sum;
+  });
+
+  this.subject({data: three_ranges});
+  this.render();
+
+  // Find the number of pixels between consecutive y-axis ticks,
+  // and the data value difference that number of pixels represents.
+  pixelsPerDataUnit = (() => {
```

Comment:
I tried to explain in the comment, but I wasn't clear enough. Let me try again:

1. We grab the first two label ticks on the Y-axis.
2. We compute the distance between the two ticks in pixels (the `transformY` stuff).
3. We compute the difference between the two numbers on the tick labels.
4. We divide the pixel difference (2) by the data value difference (3) and return the result.

That is how the function computes "pixels per data unit". Then we use this value to determine how tall each bar should be, given the gross sum of all the slice data points for that bar. Finally we assert that that expected bar height is the same as the actual bar height.